sorc/chgres_cube.fd/model_grid.F90

```diff
@@ -788,12 +796,313 @@ subroutine define_input_grid_gfs_grib2(localpet, npets)
+                          computationalLBound=clb, &
+                          computationalUBound=cub, &
+                          farrayPtr=lat_src_ptr, rc=error)
+   if(ESMF_logFoundError(rcToCheck=error,msg=ESMF_LOGERR_PASSTHRU,line=__LINE__,file=__FILE__)) &
+      call error_handler("IN GridGetCoord", error)
+
+    do j = clb(2),cub(2)
+      do i = clb(1), cub(1)
+        lon_src_ptr(i,j)=real(longitude_one_tile(i,j),esmf_kind_r8)
+        lat_src_ptr(i,j)=real(latitude_one_tile(i,j),esmf_kind_r8)
+      enddo
+    enddo
+
+   print*,"- CALL GridAddCoord FOR INPUT GRID."
+   call ESMF_GridAddCoord(input_grid, &
+                          staggerloc=ESMF_STAGGERLOC_CORNER, rc=error)
+   if(ESMF_logFoundError(rcToCheck=error,msg=ESMF_LOGERR_PASSTHRU,line=__LINE__,file=__FILE__)) &
+      call error_handler("IN GridAddCoord", error)
+
+   print*,"- CALL GridGetCoord FOR INPUT GRID X-COORD."
+   nullify(lon_src_ptr)
+   call ESMF_GridGetCoord(input_grid, &
+                          staggerLoc=ESMF_STAGGERLOC_CORNER, &
+                          coordDim=1, &
+                          farrayPtr=lon_src_ptr, rc=error)
+   if(ESMF_logFoundError(rcToCheck=error,msg=ESMF_LOGERR_PASSTHRU,line=__LINE__,file=__FILE__)) &
+      call error_handler("IN GridGetCoord", error)
+
+   print*,"- CALL GridGetCoord FOR INPUT GRID Y-COORD."
+   nullify(lat_src_ptr)
+   call ESMF_GridGetCoord(input_grid, &
+                          staggerLoc=ESMF_STAGGERLOC_CORNER, &
+                          coordDim=2, &
+                          computationalLBound=clb, &
+                          computationalUBound=cub, &
+                          farrayPtr=lat_src_ptr, rc=error)
+   if(ESMF_logFoundError(rcToCheck=error,msg=ESMF_LOGERR_PASSTHRU,line=__LINE__,file=__FILE__)) &
+      call error_handler("IN GridGetCoord", error)
+
+   print*,'bounds for corners ',localpet,clb(1),cub(1),clb(2),cub(2)
+   
+  ! If we have data on a lat/lon or lambert grid, create staggered coordinates
+  if(trim(input_grid_type)=="latlon" .or. trim(input_grid_type) == "lambert") then
+     if (trim(input_grid_type) == "latlon") then
+
+      deltalon = abs(longitude_one_tile(2,1)-longitude_one_tile(1,1))
+       do j = clb(2), cub(2)
+         do i = clb(1), cub(1)
+
+           if (i == ip1_input) then
+             lon_src_ptr(i,j) = longitude_one_tile(i_input,1) +  (0.5_esmf_kind_r8*deltalon)
+           else
+             lon_src_ptr(i,j) = longitude_one_tile(i,1) - (0.5_esmf_kind_r8*deltalon)
+           endif
+
+           if (j == jp1_input) then
+             lat_src_ptr(i,j) = latitude_one_tile(1,j_input) + (0.5_esmf_kind_r8*deltalon)
+           else
+             lat_src_ptr(i,j) = latitude_one_tile(1,j) - (0.5_esmf_kind_r8*deltalon)
+           endif
+
+         enddo
+       enddo
+     else
+       if (localpet==0) then
+         cmdline_msg = "wgrib2 "//trim(the_file)//" -d 1 -grid &> temp2.out"
+         call system(cmdline_msg)
+         open(4,file="temp2.out")
```

Comment:
I thought this was replaced by a call to grb2_inq.

Reply:
Ah, it looks like you found another use that I had forgotten about. Thanks for pointing that out. I should able to replace that relatively quickly.